dags/ecocounter_pull.py

```diff
@@ -0,0 +1,186 @@
+from airflow.models import Variable
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.macros import ds_add
+from airflow.exceptions import AirflowSkipException
+
+try:
+    repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+    sys.path.insert(0, repo_path)
+    from dags.dag_functions import task_fail_slack_alert, send_slack_msg
+    from dags.custom_operators import SQLCheckOperatorWithReturnValue
+    from volumes.ecocounter.pull_data_from_api import (
+        getToken, getSites, getChannelData, siteIsKnownToUs, insertSite,
+        insertFlow, flowIsKnownToUs, truncateFlowSince, insertFlowCounts,
+        getKnownSites, getKnownChannels
+    )
+except:
+    raise ImportError("Cannot import DAG helper functions.")
+
+DAG_NAME = 'ecocounter_pull'
+DAG_OWNERS = Variable.get('dag_owners', deserialize_json=True).get(DAG_NAME, ["Unknown"])
+
+CONFIG_PATH = '/data/airflow/data_scripts/volumes/ecocounter/.api-credentials.config'
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+#LOGGER.setLevel('DEBUG')
+
+default_args = {
+    'owner': ','.join(DAG_OWNERS),
+    'depends_on_past': False,
+    'start_date': pendulum.datetime(2024, 3, 20, tz="America/Toronto"),
+    'email_on_failure': False,
+    'email_on_success': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=5),
+    'on_failure_callback': task_fail_slack_alert
+}
+
+@dag(
+    dag_id=DAG_NAME,
+    default_args=default_args,
+    schedule='0 3 * * *',
+    template_searchpath=os.path.join(repo_path,'dags/sql'),
+    catchup=False,
+    tags=["ecocounter", "data_pull", "data_checks"],
+    doc_md=__doc__
+)
+def pull_ecocounter_dag():
+
+    @task()
+    def update_sites_and_flows(**context):
+        config = ConfigParser()
+        config.read(CONFIG_PATH)
+        token = getToken(CONFIG_PATH)
+        eco_postgres = PostgresHook("ecocounter_bot")
+
+        new_sites, new_flows = [], []
+        with eco_postgres.get_conn() as conn:
+            for site in getSites(token):
+                site_id, site_name = site['id'], site['name']
+                if not siteIsKnownToUs(site_id, conn):
+                    insertSite(conn, site_id, site_name, site['longitude'], site['latitude'])
+                    new_sites.append({
+                        'site_id': site_id,
+                        'site_name': site_name
+                    })
+                for channel in site['channels']:
+                    channel_id, channel_name = channel['id'], channel['name']
+                    if not flowIsKnownToUs(channel['id'], conn):
+                        insertFlow(conn, channel_id, site_id, channel_name, channel['interval'])
+                        new_flows.append({
+                            'site_id': site_id,
+                            'channel_id': channel_id,
+                            'flow_name': channel_name
+                        })
```

Comment:
A "channel" _is_ a "flow", right? 

This block could be a little confusing later if we forget about the incongruous terminologies. Perhaps just stick to "flow" as much as possible with the var names.